packages/compiler-sfc/src/parse.ts

```diff
@@ -95,11 +96,18 @@ export function parse(
     sourceRoot = '',
     pad = false,
     ignoreEmpty = true,
-    compiler = CompilerDOM
+    compiler = CompilerDOM,
+    defaultScriptLang
   }: SFCParseOptions = {}
 ): SFCParseResult {
   const sourceKey =
-    source + sourceMap + filename + sourceRoot + pad + compiler.parse
+    source +
+    sourceMap +
+    filename +
+    sourceRoot +
+    pad +
+    compiler.parse +
+    (defaultScriptLang || '')
```

Comment:
I would offer to use `defaultScriptLang` here without `||` and add default value while destructing SFCParseOptions instead.
This would be more consistent
<img width="597" alt="Screenshot 2022-11-19 at 10 05 42" src="https://user-images.githubusercontent.com/22274408/202841348-175dd3ee-4661-4295-8034-42ffe40f5c2d.png">

Reply:
When this option is not set, the original value of the script lang (`undefined`) may be broken by the new value `''`.

Reply:
Then we can skip it at all. Why we need to add `''` as default?

Reply:
Because the conversion of `undefined` to a string may conflict with `"undefined"`

Reply:
Possible solution
```ts
+ (defaultScriptLang != null ? `[${defaultScriptLang}]` : '')
```